storage/storage.go

```diff
@@ -52,57 +52,71 @@ type TupleKeyIterator = Iterator[*openfgapb.TupleKey]
 type uniqueObjectIterator struct {
 	iter1, iter2 ObjectIterator
 	objects      sync.Map
+	ctx          context.Context
 }
 
 // NewUniqueObjectIterator returns an ObjectIterator that iterates over two ObjectIterators and yields only distinct
 // objects with the duplicates removed.
 //
 // iter1 should generally be provided by a constrained iterator (e.g. contextual tuples) and iter2 should be provided
 // by a storage iterator that already guarantees uniqueness.
-func NewUniqueObjectIterator(iter1, iter2 ObjectIterator) ObjectIterator {
+// If the context is cancelled, Next() will return ErrIteratorDone
+func NewUniqueObjectIterator(iter1, iter2 ObjectIterator, ctx context.Context) ObjectIterator {
```

Comment:
Go standard is to pass `ctx` as the first parameter. To be more consistent with Go code in general would you mind moving it to the front?